<?xml version="1.0"?>
<launch>
<param name="isSIM" value="0"/>
    <group ns="drone1">
        <param name="isSIM" value="0"/>
        <node pkg="image_transport" type="republish" name="republish" output="screen" args="h264 in:=tello/image_raw raw out:=tello_raw"/>
        <node pkg="mtma" type="findBoxWCannyOnLine.py" name="findBoxOnLine" output="screen"/>
        <node pkg="mtma" type="control_val_node_kf.py" name="control_val_node_kf" output="screen"/>
        <node pkg="mtma" type="kf.py" name="kf" output="screen"/>
        <param name="x0" value="2"/>
        <param name="y0" value="-1"/>
        <param name="z0" value="0"/>
        <param name="th0" value="1.5708"/>
        <node pkg="mtma" type="ref_filter.py" name="ref_filter" output="screen"/>
        <node pkg="mtma" type="merge_box_node.py" name="merge_box_node" output="screen"/>
        <node pkg="mtma" type="imuint.py" name="imuint" output="screen"/>
        <node pkg="rosbag" type="record" name="myrosbag"  output="screen" args="-e '/drone1/(.*)' -x '/drone(.*)/tello_raw' -o 'drone1'" />
		<include file="$(find tello_driver)/launch/one_tello.launch">
            <arg name="net_card_name" value="wlx5c925e80f3f7"/>
        </include>
    </group>
    
    <group ns="drone2">
        <param name="isSIM" value="0"/>
        <node pkg="image_transport" type="republish" name="republish" output="screen" args="h264 in:=tello/image_raw raw out:=tello_raw"/>
        <node pkg="mtma" type="findBoxWCannyOnLine.py" name="findBoxOnLine" output="screen"/>
        <node pkg="mtma" type="control_val_node_kf.py" name="control_val_node_kf" output="screen"/>
        <node pkg="mtma" type="kf.py" name="kf" output="screen"/>
        <param name="x0" value="2"/>
        <param name="y0" value="1"/>
        <param name="z0" value="0"/>
        <param name="th0" value="1.5708"/>

        <node pkg="mtma" type="ref_filter.py" name="ref_filter" output="screen"/>
        <node pkg="mtma" type="merge_box_node.py" name="merge_box_node" output="screen"/>
        <node pkg="mtma" type="imuint.py" name="imuint" output="screen"/>
        <node pkg="rosbag" type="record" name="myrosbag"  output="screen" args="-e '/drone2/(.*)' -x '/drone(.*)/tello_raw' -o 'drone2'" />
		<include file="$(find tello_driver)/launch/one_tello.launch">
            <arg name="net_card_name" value="wlx08beac1fec81"/>
        </include>
    </group>


    <node pkg="mtma" type="target_kf.py" name="target_kf" output="screen"/>
    <!-- <node pkg="mtma" type="plan_wp.py" name="plan_wp" output="screen"/> -->
    <!-- <node pkg="mtma" type="switch_plan_wp.py" name="switch_plan_wp" output="screen"/> -->
    <node pkg="mtma" type="all_ref_node.py" name="all_ref_node" output="screen"/>
    <node pkg="mtma" type="control_op_node.py" name="control_op_node" output="screen" launch-prefix="xterm -geometry 150x20+1000+0 -e" />
    <node pkg="rosbag" type="record" name="myrosbag"  output="screen" args="-a -x '/drone(.*)/tello_raw'" />
    <!-- <node pkg="rosbag" type="record" name="myrosbag"  output="screen" args="-a" /> -->
</launch>